src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetOffencesForPersonService.kt

```diff
@@ -28,15 +28,27 @@ class GetOffencesForPersonService(
       return Response(emptyList(), responseFromProbationOffenderSearch.errors)
     }
 
-    val nomisOffences = nomisGateway.getOffencesForPerson(responseFromPrisonerOffenderSearch.data.first().identifiers.nomisNumber!!)
-    val nDeliusOffences = nDeliusGateway.getOffencesForPerson(responseFromProbationOffenderSearch.data?.identifiers?.deliusCrn!!)
+    val nomisNumber = responseFromPrisonerOffenderSearch.data.firstOrNull()?.identifiers?.nomisNumber
+    val deliusCrn = responseFromProbationOffenderSearch.data?.identifiers?.deliusCrn
 
-    if (nomisOffences.errors.isNotEmpty()) {
-      return Response(emptyList(), nomisOffences.errors)
+    var nomisOffences: Response<List<Offence>> = Response(data = emptyList())
+
+    if (nomisNumber != null) {
+      nomisOffences = nomisGateway.getOffencesForPerson(nomisNumber)
+
+      if (nomisOffences.errors.isNotEmpty()) {
+        return Response(emptyList(), nomisOffences.errors)
```

Comment:
Not sure we should be returning here, what if offences can be found in Delius?

I think we agreed to surface all the errors from the service and let the controller decide what it wants to do with them?